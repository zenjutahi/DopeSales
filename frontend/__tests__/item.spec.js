import ItemComponent from "../components/Item";
import { shallow } from "enzyme";

const fakeItem = {
  id: "adcde1234",
  title: "A Cool Item",
  price: 6500,
  description: "This item is so cool it drives me crazy",
  image: "item.jpg",
  largeImage: "item-large.jpg"
};

describe("<Item/>", () => {
  const wrapper = shallow(<ItemComponent item={fakeItem} />);
  it("renders the image properly", () => {
    // const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const img = wrapper.find("img");
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it("renders and displays title and price properly", () => {
    // const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const priceTag = wrapper.find("PriceTag");
    const title = wrapper.find("Title a");
    expect(priceTag.children().text()).toBe("$65");
    expect(title.text()).toBe(fakeItem.title);
  });

  it("renders out the buttons properly", () => {
    // const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find(".buttonList");
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find("Link")).toBeTruthy();
    expect(buttonList.find("AddToCart")).toHaveLength(1);
  });
});
