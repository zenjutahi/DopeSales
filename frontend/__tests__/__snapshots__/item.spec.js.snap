// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders the Item snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "adcde1234",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $65
  </PriceTag>
  <p>
    This item is so cool it drives me crazy
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "adcde1234",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="adcde1234"
    />
    <DeleteItem
      id="adcde1234"
    >
      Delete 🗑️
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
